### ISSUE TITLE
ami_analyze takes forever to run on full-frame exposures

### ISSUE THREAD

**Issue by hbushouse** [STScI-JWST/jwst#1837]
While DMS was testing `calwebb_ami3` on some data (from who knows where) they ran across a situation where the pipeline ran for more than 24 hours before someone finally just killed it. Turns out it was trying to process one or more full-frame AMI exposures, which I've confirmed take a really long time to process through the `ami_analyze` step. All of the test data we've used during development (provided by the AMI WG) has been much smaller 19x19 or 80x80 subarrays. I've confirmed that when I take an 80x80 subarray out of the middle of the original full-frame exposure, it then only takes ~7 minutes to process through `ami_analyze`.

I've queried the AMI WG as to whether we should always extract an 80x80 subarray from full-frame data when sending it through `ami_analyze`. It's doubtful that there would ever be useful source information outside of that region and hence shouldn't affect the processing. Waiting for the AMI WG to respond.

An example test exposure that causes the problem is jw00308001001_02102_00001_nis_cal.fits

**Comment by hbushouse** [comment]
The AMI WG agrees that it's OK and appropriate to extract an 80x80 subarray from full-frame exposures (equivalent to the user using the SUB80 subarray in the first place) before sending the data into the ami_analyze fringe computation algorithms.

Extraction of the subarray could probably be implemented in the ami_analyze.apply_LG function somewhere before the image is sent to the jwnrm.fit_image() function.

**Comment by hbushouse** [comment]
The AMI observing template in APT allows a user to select a SUBARRAY of "FULL" or "SUB80". So our extraction from full-frame exposures should match the location and size of the SUB80 subarray. The latest info we have for the SUB80 characteristics is:
```
SUBARRAY= 'SUB80   '           / Subarray used                                  
SUBSTRT1=                 1045 / Starting pixel in axis 1 direction             
SUBSTRT2=                    1 / Starting pixel in axis 2 direction             
SUBSIZE1=                   80 / Number of pixels in axis 1 direction           
SUBSIZE2=                   80 / Number of pixels in axis 2 direction  
```
which is based on the most recent dark and superbias ref files in CRDS that use the SUB80 subarray.

**Comment by hbushouse** [comment]
Fixed by #2017
